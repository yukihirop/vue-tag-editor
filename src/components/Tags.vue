<template>
  <span>
    <span v-for='(tag, index) in tags'>
      <tag-label v-if='isLabel' :tagname='tag' @delete-tag='deleteTag(index)'></tag-label>
      <tag-link v-if='isLink' :tagname='tag' @delete-tag='deleteTag(index)'></tag-link>
    </span>
  </span>
</template>

<script>
import TagLabel from "./tags/TagLabel.vue"
import TagLink from "./tags/TagLink.vue"

export default {
  name: "tags",
  props: {
    tags: {
      type: Array,
      default: []
    },
    type: {
      type: String,
      default: 'label'
    }
  },
  components: {
    TagLabel: TagLabel,
    TagLink: TagLink
  },
  methods: {
    deleteTag(index){
      this.tags.splice(index, 1)
    }
  },
  computed:{
    isLabel: function() {
      return this.type === 'label'
    },
    isLink: function() {
      return this.type === 'link'
    }
  }
}
</script>