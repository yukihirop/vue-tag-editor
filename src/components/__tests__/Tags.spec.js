import { createLocalVue, shallowMount } from '@vue/test-utils'
import Tags from '../Tags'

const localVue = createLocalVue()

describe('Tags', () => {
  const wrapper = shallowMount(Tags,{
    localVue,
    propsData:{
      tags: ['javascript', 'ruby']
    }
  })

  it('correct name options', () => {
    expect(wrapper.vm.$options.name).toBe('tags')
  })

  it('correct default props', () => {
    expect(wrapper.props().type).toBe('')
  })

  it('renders correctly (snapshots)', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  describe('renders the correct markup', () => {
    describe('when type is label', () => {
      beforeEach(() => {
        wrapper.setProps({
          type: 'label'
        })
      })

      it('renders the tag-label component', () => {
        expect(wrapper.html()).toContain("<tag-label-stub tagname=\"javascript\"></tag-label-stub>")
        expect(wrapper.html()).toContain("<tag-label-stub tagname=\"ruby\"></tag-label-stub>")
      })
    })

    describe('when type is link', () => {
      beforeEach(() => {
        wrapper.setProps({
          type: 'link'
        })
      })

      it('renders the tag-link component', () => {
        expect(wrapper.html()).toContain("<tag-link-stub tagname=\"javascript\"></tag-link-stub>")
        expect(wrapper.html()).toContain("<tag-link-stub tagname=\"ruby\"></tag-link-stub>")
      })
    })
  })
})