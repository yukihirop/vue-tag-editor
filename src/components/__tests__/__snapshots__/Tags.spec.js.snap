// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags renders correctly (snapshots) 1`] = `
<span>
  <span>
    <!---->
     
    <!---->
  </span>
  <span>
    <!---->
     
    <!---->
  </span>
</span>
`;
